managers/restore-manager/RestoreService.js

```diff
@@ -0,0 +1,297 @@
+              //following can be treated as extra processing
+              // just to avoid duplicate entries in restore histroy
+              // which might lead to quota full
+              let uniqueDates = [...new Set(restoreDatesByState)];
+              const patchObj = {
+                state: lastOperation.state,
+                logs: logs,
+                finished_at: restoreFinishiedAt,
+                restore_dates: _.chain(restoreDates)
+                  .set(lastOperation.state, _.sortBy(uniqueDates))
+                  .value()
+              };
+              return this.backupStore
+                .patchRestoreFile(options, patchObj)
+                .then(() => eventmesh.apiServerClient.patchResource({
+                  resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.RESTORE,
+                  resourceType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_RESTORE,
+                  resourceId: opts.restore_guid,
+                  status: {
+                    response: patchObj
+                  }
+                }));
+            })
+            .tap(() => {
+              // Trigger schedule backup when restore is successful
+              // statusAlreadyChecked should be false, otherwise
+              // if by chance restore state (this function called) two times
+              // for any reason after agent restores successfully,
+              // on seceond time also it would trigger schedule backup (which is incorrect).
+              if (lastOperation.state === CONST.OPERATION.SUCCEEDED &&
+                !statusAlreadyChecked) {
+                return this.reScheduleBackup({
+                  instance_id: options.instance_guid,
+                  afterXminute: config.backup.reschedule_backup_delay_after_restore || 3
+                });
+              } else {
+                logger.debug(`Not re-scheduling backup for ${options.instance_guid} as current restore state check is not first time.`);
+                return;
+              }
+            });
+        }
+      });
+  }
+
+  reScheduleBackup(opts) {
+    const options = {
+      instance_id: opts.instance_id,
+      repeatInterval: 'daily',
+      type: CONST.BACKUP.TYPE.ONLINE
+    };
+
+    if (this.plan.service.backup_interval) {
+      options.repeatInterval = this.plan.service.backup_interval;
+    }
+
+    options.repeatInterval = utils.getCronWithIntervalAndAfterXminute(options.repeatInterval, opts.afterXminute);
+    logger.info(`Scheduling Backup for instance : ${options.instance_id} with backup interval of - ${options.repeatInterval}`);
+    //Even if there is an error while fetching backup schedule, trigger backup schedule we would want audit log captured and riemann alert sent
+    return retry(() => cf.serviceFabrikClient.scheduleBackup(options), {
+      maxAttempts: 3,
+      minDelay: 500
+    });
+  }
+
+  startRestore(opts) {
+    logger.debug('Starting restore with options:', opts);
+    const deploymentName = opts.deployment;
```

Comment:
Is it possible to get deployment name from manager itself rather than controller find deploymentname `findDeploymentNameByInstanceId`?